### DISCUSSION TITLE
Typescript References

### DISCUSSION THREAD

**Discussion by mxkyb** [callstack/react-native-builder-bob#499] [Q&A]
Hello everyone,
how do you handle typescript references with react-native projects? For example something similar to this:

project1/tsconfig.json:
```json
{
    "compilerOptions": {
        "composite": true
    },
    "include": ["src/**/*"]
}
```

project2/tsconfig.json:
```json
{
    "references": [
        { "path": "../project1" }
    ],
    "include": ["src"]
}
```

Now if we'd run `bob` on project2, project1 wouldn't be bundled into the source files in the experiments I made. A problem I see is that we can't bundle all the sources together, because we need to preserve react-natives platform resolution file extensions like `.android`, `.ios`, ...
Appreciate any ideas :)

**Comment by satya164** [comment]
If they are libraries in a monorepo and depend on each other, then tools like `lerna` take care of running the builds in the appropriate order when publishing.
